breadbox/breadbox/logging.py

```diff
@@ -0,0 +1,73 @@
+import logging.config
+import logging
+from typing import Optional
+
+from fastapi import Request
+from google.cloud import error_reporting
+from pydantic import BaseModel
+
+# based on https://stackoverflow.com/questions/63510041/adding-python-logging-to-fastapi-endpoints-hosted-on-docker-doesnt-display-api
+class LogConfig(BaseModel):
+    LOGGER_NAME: str = "breadbox"
+    LOG_FORMAT: str = "%(levelprefix)s | %(asctime)s | %(message)s"
+    LOG_LEVEL: str = "INFO"
+
+    # Logging config
+    version: int = 1
+    disable_existing_loggers: bool = False
+    formatters: dict = {
+        "default": {
+            "()": "uvicorn.logging.DefaultFormatter",
+            "fmt": LOG_FORMAT,
+            "datefmt": "%Y-%m-%d %H:%M:%S",
+        },
+    }
+    handlers: dict = {
+        "default": {
+            "formatter": "default",
+            "class": "logging.StreamHandler",
+            "stream": "ext://sys.stderr",
+        },
+    }
+    loggers: dict = {
+        LOGGER_NAME: {"handlers": ["default"], "level": LOG_LEVEL},
+    }
+
+
+def configure_logging():
+    logging.config.dictConfig(LogConfig().dict())
+
+
+class GCPExceptionReporter:
+    def __init__(self, breadbox_env: str):
+        self.service_name = "breadbox-" + breadbox_env
+        self.client = self._create_client() if not breadbox_env == "dev" else None
+
+    @property
+    def disabled(self):
+        return self.client is None
+
+    def _create_client(self):
+        return error_reporting.Client(service=self.service_name)
+
+    def _create_http_context(self, request: Request, status_code: int):
+        breadbox_http_context = error_reporting.HTTPContext(
+            method=request.scope["method"],
+            url=request.scope["root_path"] + request.scope["path"],
+            response_status_code=status_code,
+        )
+        return breadbox_http_context
+
+    def report(self, request: Optional[Request], status_code: Optional[int], user: Optional[str]):
+        if self.client is None:
+            print("Error reporting disabled")
```

Comment:
So far, I've just been running this in debug mode and logging the stacktrace. On monday, I'll try these changes out in IQA to make sure things are working with all of the real configurations.